Importer.pm

```diff
@@ -247,36 +249,39 @@ sub needsUpdate { if (!main::SCANNER) {
 	my ($class, $cb) = @_;
 
 	my $lastScanTime = $cache->get('tidal_library_last_scan') || return $cb->(1);
+	my @keys = qw(updatedFavoriteArtists updatedFavoriteTracks updatedFavoriteAlbums);
+	push @keys, qw(updatedFavoritePlaylists updatedPlaylists) unless $class->ignorePlaylists;
 
-	my $checkFav = sub {
-		my ($userId, $type, $previous, $acb) = @_;
+	Async::Util::achain(
+		steps => [ map {
+			my $userId = $_;
+			sub {
+				my ($input, $acb) = @_;
+				return $acb($input) if $input;
 
-		Plugins::TIDAL::API::Async->new({
-			userId => $userId
-		})->getLatestCollectionTimestamp(sub {
-			my $timestamp = shift;
-			$acb->($timestamp > $lastScanTime);
-		}, $type);
-	};
+				my $api = Plugins::TIDAL::API::Async->new({ userId => $userId });
 
-	my $workers = [ map {
-		my $userId = $_;
-		my @tasks = (
-			sub { $checkFav->($userId, 'albums', @_) },
-			sub { $checkFav->($userId, 'artists', @_) },
-		);
+				$api->getLatestCollectionTimestamp( sub {
+					my (undef, $timestamp) = @_;
 
-		if (!$class->ignorePlaylists) {
-			push @tasks, sub { $checkFav->($userId, 'playlists', @_) };
-		}
+					foreach (@keys) {
+						return $acb->(1) if $timestamp->{$_} > $lastScanTime;
+					}
 
-		@tasks;
-	} sort {
-		$a <=> $b
-	} values %{_enabledAccounts()} ];
+					return $acb(0) if $class->ignorePlaylists;
 
-	Async::Util::achain(
-		steps => $workers,
+					$api->getFavoritePlaylists( sub {
+						my $playlists = shift;
+
+						foreach (@$playlists) {
+							return $acb->(1) if str2time($_->{item}->{lastUpdated}) > $lastScanTime;
+						}
+
+						return $acb->(0);
+					}, 'playlists', 1 );
```

Comment:
`getFavoritePlaylists()` would only accept one param in addition to the callback. I guess the `playlists` is a left over from calling `getFavorites()`?